### ISSUE TITLE
Can't send image to the RP

### ISSUE THREAD

**Issue by anriijmind** [reportportal/agent-android#12]
I try to add a screenshot/image to the report and have an error in the Logger class.

I have logger utils calls:
```
object LoggingUtils {
    private val LOGGER: Logger = LoggerFactory.getLogger("binary_data_logger")

    fun log(file: String, message: String?) {
        LOGGER.info("RP_MESSAGE#FILE#{}#{}", file, message)
    }

    fun log(bytes: ByteArray?, message: String?) {
        LOGGER.info("RP_MESSAGE#BASE64#{}#{}", Base64.getEncoder().encodeToString(bytes), message)
    }

    fun logBase64(base64: String?, message: String?) {
        LOGGER.info("RP_MESSAGE#BASE64#{}#{}", base64, message)
    }
}
```

In the test  presence code for sending file:

```
val targetContext = InstrumentationRegistry.getInstrumentation().targetContext
val file = EspressoScreenshot.takeScreenshotPath("rp-test8303202978285528238.png", targetContext)
Log.i("Logi", File(file).exists().toString())
LoggingUtils.log(file, "Android launch with file")
```

So file locates at this directory: /data/user/0/com.epam.test/cache/rp-test8303202978285528238.png
2021-04-12 11:52:56.943 16833-16869/com.epam.test I/Logi: true

And I have that error:
```
java.lang.ExceptionInInitializerError
	at org.apache.tika.mime.MimeTypesFactory.create(MimeTypesFactory.java:69)
	at org.apache.tika.mime.MimeTypesFactory.create(MimeTypesFactory.java:100)
	at org.apache.tika.mime.MimeTypesFactory.create(MimeTypesFactory.java:189)
	at org.apache.tika.mime.MimeTypes.getDefaultMimeTypes(MimeTypes.java:604)
	at org.apache.tika.config.TikaConfig.getDefaultMimeTypes(TikaConfig.java:82)
	at org.apache.tika.config.TikaConfig.<init>(TikaConfig.java:247)
	at org.apache.tika.config.TikaConfig.getDefaultConfig(TikaConfig.java:386)
	at org.apache.tika.parser.AutoDetectParser.<init>(AutoDetectParser.java:55)
	at com.epam.reportportal.utils.MimeTypeDetector.<clinit>(MimeTypeDetector.java:37)
	at com.epam.reportportal.utils.MimeTypeDetector.detect(MimeTypeDetector.java:57)
	at com.epam.reportportal.utils.files.Utils.getFile(Utils.java:131)
	at com.epam.reportportal.message.HashMarkSeparatedMessageParser$MessageType$1.toByteSource(HashMarkSeparatedMessageParser.java:54)
	at com.epam.reportportal.message.HashMarkSeparatedMessageParser.parse(HashMarkSeparatedMessageParser.java:116)
	at com.epam.reportportal.logback.appender.ReportPortalAppender.lambda$append$0$ReportPortalAppender(ReportPortalAppender.java:55)
	at com.epam.reportportal.logback.appender.-$$Lambda$ReportPortalAppender$h9OyensytHEmozo0ozSSnM19Lbo.apply(Unknown Source:6)
	at com.epam.reportportal.service.LoggingContext.prepareRequest(LoggingContext.java:131)
	at com.epam.reportportal.service.LoggingContext.lambda$emit$1$LoggingContext(LoggingContext.java:148)
	at com.epam.reportportal.service.-$$Lambda$LoggingContext$K_IOQDBW1XzYIx0WSWzjuqV5Ea4.apply(Unknown Source:8)
	at io.reactivex.internal.functions.Functions$Array2Func.apply(Functions.java:529)
	at io.reactivex.internal.functions.Functions$Array2Func.apply(Functions.java:516)
	at io.reactivex.internal.operators.maybe.MaybeZipArray$ZipCoordinator.innerSuccess(MaybeZipArray.java:111)
	at io.reactivex.internal.operators.maybe.MaybeZipArray$ZipMaybeObserver.onSuccess(MaybeZipArray.java:176)
	at io.reactivex.internal.operators.maybe.MaybeCache.subscribeActual(MaybeCache.java:66)
	at io.reactivex.Maybe.subscribe(Maybe.java:4290)
	at io.reactivex.internal.operators.maybe.MaybeZipArray.subscribeActual(MaybeZipArray.java:62)
	at io.reactivex.Maybe.subscribe(Maybe.java:4290)
	at io.reactivex.internal.operators.maybe.MaybeToFlowable.subscribeActual(MaybeToFlowable.java:45)
	at io.reactivex.Flowable.subscribe(Flowable.java:14918)
	at io.reactivex.Flowable.subscribe(Flowable.java:14865)
	at io.reactivex.internal.operators.flowable.FlowableFlatMap$MergeSubscriber.onNext(FlowableFlatMap.java:163)
	at io.reactivex.internal.operators.flowable.FlowableOnBackpressureBuffer$BackpressureBufferSubscriber.drain(FlowableOnBackpressureBuffer.java:187)
	at io.reactivex.internal.operators.flowable.FlowableOnBackpressureBuffer$BackpressureBufferSubscriber.onNext(FlowableOnBackpressureBuffer.java:112)
	at io.reactivex.internal.operators.flowable.FlowableFromObservable$SubscriberObserver.onNext(FlowableFromObservable.java:54)
	at io.reactivex.subjects.PublishSubject$PublishDisposable.onNext(PublishSubject.java:308)
	at io.reactivex.subjects.PublishSubject.onNext(PublishSubject.java:228)
	at com.epam.reportportal.service.LoggingContext.emit(LoggingContext.java:148)
	at com.epam.reportportal.service.ReportPortal.emitLog(ReportPortal.java:218)
	at com.epam.reportportal.logback.appender.ReportPortalAppender.append(ReportPortalAppender.java:43)
	at com.epam.reportportal.logback.appender.ReportPortalAppender.append(ReportPortalAppender.java:36)
	at ch.qos.logback.core.AppenderBase.doAppend(AppenderBase.java:82)
	at ch.qos.logback.core.spi.AppenderAttachableImpl.appendLoopOnAppenders(AppenderAttachableImpl.java:51)
	at ch.qos.logback.classic.Logger.appendLoopOnAppenders(Logger.java:270)
	at ch.qos.logback.classic.Logger.callAppenders(Logger.java:257)
	at ch.qos.logback.classic.Logger.buildLoggingEventAndAppend(Logger.java:421)
	at ch.qos.logback.classic.Logger.filterAndLog_2(Logger.java:414)
	at ch.qos.logback.classic.Logger.info(Logger.java:587)
	at com.epam.test.android.espresso.junit5.LoggingUtils.log(LoggingUtils.kt:12)
	at com.epam.test.android.espresso.junit5.UiTest.test_short_password_error_1(UiTest.kt:88)
	at java.lang.reflect.Method.invoke(Native Method)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at android.com.epam.reportportal.junit5.ReportPortalExtension.interceptTestMethod(ReportPortalExtension.java:222)
	at org.junit.jupiter.engine.descriptor.-$$Lambda$gvLh9pTap0g17ZemJcSAjpfE7To.apply(Unknown Source:0)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.-$$Lambda$ExecutableInvoker$ReflectiveInterceptorCall$AfiRfaQW5MFAa8lovVwndaKa8l0.apply(Unknown Source:2)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.-$$Lambda$ExecutableInvoker$hmdF2IKSQSF2OJBv_amWUUg0sS8.apply(Unknown Source:6)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.descriptor.-$$Lambda$gvLh9pTap0g17ZemJcSAjpfE7To.apply(Unknown Source:0)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.-$$Lambda$ExecutableInvoker$ReflectiveInterceptorCall$AfiRfaQW5MFAa8lovVwndaKa8l0.apply(Unknown Source:2)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.-$$Lambda$ExecutableInvoker$hmdF2IKSQSF2OJBv_amWUUg0sS8.apply(Unknown Source:6)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6$TestMethodTestDescriptor(TestMethodTestDescriptor.java:210)
	at org.junit.jupiter.engine.descriptor.-$$Lambda$TestMethodTestDescriptor$HI566US9RrpykSmg1FzjlYD6qrA.execute(Unknown Source:6)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5$NodeTestTask(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.-$$Lambda$NodeTestTask$oEtjVUBr1dzvAyO4rHrqPrrp8iU.execute(Unknown Source:2)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7$NodeTestTask(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.-$$Lambda$NodeTestTask$sIF-wJyqQcSlbWMGWi7dBYWbuCo.invoke(Unknown Source:2)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8$NodeTestTask(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.-$$Lambda$NodeTestTask$EAUqZA5CfD8zbN3AlxhbCoS66eU.execute(Unknown Source:2)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5$NodeTestTask(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.-$$Lambda$NodeTestTask$oEtjVUBr1dzvAyO4rHrqPrrp8iU.execute(Unknown Source:2)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7$NodeTestTask(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.-$$Lambda$NodeTestTask$sIF-wJyqQcSlbWMGWi7dBYWbuCo.invoke(Unknown Source:2)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8$NodeTestTask(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.-$$Lambda$NodeTestTask$EAUqZA5CfD8zbN3AlxhbCoS66eU.execute(Unknown Source:2)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService.invokeAll(ForkJoinPoolHierarchicalTestExecutorService.java:129)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5$NodeTestTask(NodeTestTask.java:143)
	at org.junit.platform.engine.support.hierarchical.-$$Lambda$NodeTestTask$oEtjVUBr1dzvAyO4rHrqPrrp8iU.execute(Unknown Source:2)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7$NodeTestTask(NodeTestTask.java:129)
	at org.junit.platform.engine.support.hierarchical.-$$Lambda$NodeTestTask$sIF-wJyqQcSlbWMGWi7dBYWbuCo.invoke(Unknown Source:2)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8$NodeTestTask(NodeTestTask.java:127)
	at org.junit.platform.engine.support.hierarchical.-$$Lambda$NodeTestTask$EAUqZA5CfD8zbN3AlxhbCoS66eU.execute(Unknown Source:2)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at org.junit.platform.engine.support.hierarchical.ForkJoinPoolHierarchicalTestExecutorService$ExclusiveTask.compute(ForkJoinPoolHierarchicalTestExecutorService.java:185)
	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:285)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1155)
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1993)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1941)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.RuntimeException: problem initializing SAXParser pool
	at org.apache.tika.mime.MimeTypesReader.<clinit>(MimeTypesReader.java:119)
	... 123 more
Caused by: org.apache.tika.exception.TikaException: prooblem creating SAX parser factory
	at org.apache.tika.mime.MimeTypesReader.newSAXParser(MimeTypesReader.java:394)
	at org.apache.tika.mime.MimeTypesReader.setPoolSize(MimeTypesReader.java:378)
	at org.apache.tika.mime.MimeTypesReader.<clinit>(MimeTypesReader.java:117)
	... 123 more
Caused by: org.xml.sax.SAXNotRecognizedException: http://javax.xml.XMLConstants/feature/secure-processing
	at org.apache.harmony.xml.parsers.SAXParserFactoryImpl.setFeature(SAXParserFactoryImpl.java:93)
	at org.apache.tika.mime.MimeTypesReader.newSAXParser(MimeTypesReader.java:390)
	... 125 more
```

At the RP I see launch with that filed test. This code for adding screenshots works on my web project but fails with android - what do I do wrong? 

The project I've got from https://github.com/reportportal/android-kotlin-example

**Comment by HardNorth** [comment]
@anriijmind We use Apache Tika to guess binary data MIME type. It worked well for all kinds of java machines before and our users usually work not that good,  sending us incorrect MIME types and constantly open new issues about missed screenshots. Unfortunately Tika uses SAX parser for its internal needs and bypasses `http://javax.xml.XMLConstants/feature/secure-processing` option to it to prevent XXE attacks, but this option is not supported for Android.

I'll try to find better solution but for now please manually downgrade Apache Tika to this:
`implementation 'org.apache.tika:tika-core:1.12'`

The whole JUnit 5 dependency should look like this:
```none
    // Report Portal libraries
    implementation ('com.epam.reportportal:agent-android-junit5:5.1.0-BETA-5') {
        exclude group: 'org.aspectj' // AspectJ is already included by Android
        exclude module: 'tika-core'
    }
    implementation 'org.apache.tika:tika-core:1.12'
```

**Comment by anriijmind** [comment]
@HardNorth thank you, it works. I'm really glad about your quick answer.

**Comment by HardNorth** [comment]
Fixed in [5.1.0-RC-2](https://github.com/reportportal/agent-android/releases/tag/5.1.0-RC-2)
